### PR TITLE
fix(suite): graph causes render errors

### LINKED ISSUE #2724: Graph causes render errors
Screenshot 2020-10-22 at 19 40 01

### PR DESCRIPTION
**Pull request by tomexx**
A quick fix for a weird error from #2724

### CODE REVIEW

File: packages/suite/src/components/suite/TransactionsGraph/components/CustomTooltipBase/index.tsx [hunk 1]
```diff
@@ -120,13 +120,15 @@ const CustomTooltipBase = (props: Props) => {
...
-        }
+        useEffect(() => {
+            if (props.onShow && props.extendedDataForInterval) {
+                props.onShow(
+                    props.extendedDataForInterval.findIndex(
+                        item => item.time === props.payload![0].payload.time,
+                    ),
+                );
+            }
+        }, [])
```
Comment (keraf) [inline packages/suite/src/components/suite/TransactionsGraph/components/CustomTooltipBase/index.tsx]: Not sure about this one, but shouldn't `extendedDataForInterval` be in the dependencies?

Comment (tomexx) [reply packages/suite/src/components/suite/TransactionsGraph/components/CustomTooltipBase/index.tsx]: I'll check

Comment (tomexx) [reply packages/suite/src/components/suite/TransactionsGraph/components/CustomTooltipBase/index.tsx]: The thing is, I pass setState to a child component and it clashes with rendering of itself. I googled for a solution and found this https://stackoverflow.com/a/63289170/872981 - I applied useEffect and it works correctly now.

However you were right about the deps and props, I added those and TS works OK now. 👍

File: packages/suite/src/components/suite/TransactionsGraph/index.tsx [hunk 2]
```diff
@@ -193,7 +193,7 @@ const TransactionsGraph = React.memo((props: Props) => {
                                             receivedValueFn={props.receivedValueFn}
                                             balanceValueFn={props.balanceValueFn}
                                             extendedDataForInterval={extendedDataForInterval}
-                                            onShow={index => setHovered(index)}
+                                            onShow={(index: HoveredIndex) => setHovered(index)}
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/TransactionsGraph/index.tsx]: why do we even use this custom typing `HoveredIndex`? shouldn't it be just a `number` ?

Comment (tomexx) [reply packages/suite/src/components/suite/TransactionsGraph/index.tsx]: Originally I thought I'd use `undefined` as a part of the type, but then I switched to `-1` for the default state so the `number` type is enough. Fixed 👍

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**